nautobot/core/api/metadata.py

```diff
@@ -223,8 +224,17 @@ def get_list_display_fields(self, serializer):
 
     def get_advanced_tab_fields(self, serializer):
         """Try to get the advanced tab fields or default to an empty list."""
-        default_advanced_fields = ["id", "url", "composite_key", "created", "last_updated"]
-        advanced_fields = [field for field in default_advanced_fields if field in serializer.fields.keys()]
+        if hasattr(serializer.Meta, "advanced_view_config"):
+            # Gather all fields defined in the custom `advanced_view_config` if there is one defined on the serializer.
+            advanced_view_layout = serializer.Meta.advanced_view_config["layout"]
+            advanced_fields = []
+            for section in advanced_view_layout:
+                for value in section.values():
+                    advanced_fields += value["fields"]
+        else:
+            # Default advanced fields
+            advanced_fields = ["id", "url", "composite_key", "created", "last_updated"]
```

Comment:
This appears to be defined twice. Once here and once in the return value for `get_default_advanced_view_config()`. Is there a way we can reduce this to a single definition, like as a list stored as a class attribute?

Reply:
Stored as a class attribute